src/event/v10/attributes.rs

```diff
@@ -15,6 +15,60 @@ pub struct Attributes {
     pub(crate) time: Option<DateTime<Utc>>,
 }
 
+impl<'a> IntoIterator for &'a Attributes {
+    type Item = (&'a str, AttributeValue<'a>);
+    type IntoIter = AttributesIntoIterator<'a>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        AttributesIntoIterator {
+            attributes: self,
+            index: 0,
+        }
+    }
+}
+
+pub struct AttributesIntoIterator<'a> {
+    attributes: &'a Attributes,
+    index: usize,
+}
+
+fn option_checker_string<'a>(attribute_type: &'a str,input:Option<&'a str>) -> Option<(&'a str,AttributeValue<'a>)> {
+    let result = match input {
+        Some(x) => Some((attribute_type,AttributeValue::String(x))),
+        None => None,
+    };
+    result
+}
+
+fn option_checker_time<'a>(attribute_type: &'a str,input:Option<&'a DateTime<Utc>>) -> Option<(&'a str,AttributeValue<'a>)> {
+    let result = match input {
+        Some(x) => Some((attribute_type,AttributeValue::Time(x))),
+        None => None,
+    };
+    result
+}
+
+impl<'a> Iterator for AttributesIntoIterator<'a> {
+    type Item = (&'a str, AttributeValue<'a>);
+    fn next(&mut self) -> Option<Self::Item> {
+        let result = match self.index {
+            0 => Some(("id", AttributeValue::String(&self.attributes.id))),
+            1 => Some(("ty", AttributeValue::String(&self.attributes.ty))),        
+            2 => Some(("source", AttributeValue::String(&self.attributes.source))),
+            3 => option_checker_string("datacontenttype",self.attributes.get_datacontenttype()), 
```

Comment:
more than having `option_checker_string`, you can improve this code with:

```
self.attributes.datacontenttype.as_ref().map(|v| ("datacontenttype", AttributeValue::String(v))
```

Reply:
it's more explicative, and you pay two function invocations less (one per the getter and one per the `option_checker_string`

Reply:
One question though. Wouldn't running `.map()` on the `.as_ref()` imply that `v` is an `Option<&T>` type?